partner_statement/tests/test_activity_statement.py

```diff
@@ -10,26 +10,36 @@
 class TestActivityStatement(TransactionCase):
     """Tests for Activity Statement."""
 
-    def setUp(self):
-        super().setUp()
-
-        self.res_users_model = self.env["res.users"]
-        self.company = self.env.ref("base.main_company")
-        self.company.external_report_layout_id = self.env.ref(
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(
+            context=dict(
+                cls.env.context,
+                mail_create_nolog=True,
+                mail_create_nosubscribe=True,
+                mail_notrack=True,
+                no_reset_password=True,
+                tracking_disable=True,
+            )
+        )
+        cls.res_users_model = cls.env["res.users"]
+        cls.company = cls.env.ref("base.main_company")
+        cls.company.external_report_layout_id = cls.env.ref(
             "web.external_layout_standard"
         )
-        self.partner1 = self.env.ref("base.res_partner_1")
-        self.partner2 = self.env.ref("base.res_partner_2")
-        self.g_account_user = self.env.ref("account.group_account_user")
-
-        self.user = self._create_user("user_1", [self.g_account_user], self.company).id
-
-        self.statement_model = self.env["report.partner_statement.activity_statement"]
-        self.wiz = self.env["activity.statement.wizard"]
-        self.report_name = "partner_statement.activity_statement"
-        self.report_name_xlsx = "p_s.report_activity_statement_xlsx"
-        self.report_title = "Activity Statement"
-        self.today = fields.Date.context_today(self.wiz)
+        cls.partner1 = cls.env.ref("base.res_partner_1")
+        cls.partner2 = cls.env.ref("base.res_partner_2")
+        cls.g_account_user = cls.env.ref("account.group_account_user")
+
+        cls.user = cls._create_user("user_1", [cls.g_account_user], cls.company).id
```

Comment:
Remove the method `_create_user` and use:

```
from odoo.tests import new_test_user
```

and call it this way:

```
cls.user = new_test_user(cls.env, login="user_1", groups="account.group_account_user")
```

Do the same for each of the custom "user creation".